src/generic/codec.rs

```diff
@@ -132,17 +135,19 @@ impl<T, U> Encode<T, U>
 where T: Encoder<Item = U::Item>,
       U: Stream,
 {
-    pub(crate) fn new(encoder: T, inner: U) -> Self {
+    pub(crate) fn new(encoder: T, inner: U, return_trailers: bool) -> Self {
         Encode {
             inner: EncodeInner::Ok { encoder, inner },
             buf: BytesMut::new(),
+            return_trailers,
         }
     }
 
-    pub(crate) fn error(status: Status) -> Self {
+    pub(crate) fn error(status: Status, return_trailers: bool) -> Self {
```

Comment:
Considering that the only way to send an error is via trailers, this can probably leave the argument off and just always be true.

Reply:
Good point.